plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -73,12 +74,13 @@ static void addLibraries(IProject project, List<Library> libraries, IProgressMon
     }
     javaProject.setRawClasspath(newRawClasspath, monitor);
   
-    runContainerResolverJob(javaProject);
+    IEclipseContext context = EclipseContextFactory.getServiceContext(
+        FrameworkUtil.getBundle(classFromBundle).getBundleContext());
+    
+    runContainerResolverJob(javaProject, context);
   }
   
-  private static void runContainerResolverJob(IJavaProject javaProject) {
-    IEclipseContext context = EclipseContextFactory.getServiceContext(
-        FrameworkUtil.getBundle(CreateAppEngineStandardWtpProject.class).getBundleContext());
+  private static void runContainerResolverJob(IJavaProject javaProject, IEclipseContext context) {
```

Comment:
@chanseokoh's right: we don't need the class.  Just get the service context for this class' bundle.  I would revert this code to how it was.

The class/bundle is only important as it determines which bundle's lifecycle will be used to clean up any referenced OSGi services.